main/src/ca/uwaterloo/flix/language/phase/sjvm/Instructions.scala

```diff
@@ -20,12 +20,32 @@ import ca.uwaterloo.flix.language.ast.ERefType._
 import ca.uwaterloo.flix.language.ast.EType._
 import ca.uwaterloo.flix.language.ast.PRefType._
 import ca.uwaterloo.flix.language.ast.PType._
-import ca.uwaterloo.flix.language.ast.{Cat1, Cat2, EType, PRefType, PType, SourceLocation, Symbol}
+import ca.uwaterloo.flix.language.ast.{Cat1, Cat2, ERefType, EType, PRefType, PType, SourceLocation, Symbol}
 import ca.uwaterloo.flix.language.phase.sjvm.BytecodeCompiler._
 import ca.uwaterloo.flix.util.InternalCompilerException
 import org.objectweb.asm.Opcodes
 
 object Instructions {
+  val classStrings: scala.collection.mutable.Map[EType[_], String] = scala.collection.mutable.Map()
+  val descriptorStrings: scala.collection.mutable.Map[EType[_], String] = scala.collection.mutable.Map()
```

Comment:
This kind of structure is generally going to be problematic because:

1. We want these things to run in parallel.
2. We can have multiple instances of the compiler (which would then share these fields- which would fail).

Instead what we try to do is to decorate the ASTs with the right information. Or we pass arguments around.

Reply:
This is hard to do with object and null for example but ill just make it work and then make it better later.

Reply:
What do you mean?

Reply:
In places where null/unit/context is needed when generating code there isn't necessarily a unit expression present. Maybe they can be stored elsewhere.